File: lms-backend/src/main/java/com/lms/entity/Submission.java
package com.lms.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "submissions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Submission {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "assignment_id", nullable = false)
	private Assignment assignment;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "student_id", nullable = false)
	private User student;

	@Column(nullable = false)
	private String filePath;

	@Column(name = "submitted_at", nullable = false)
	private LocalDateTime submittedAt;

	@Column(name = "marked_at")
	private LocalDateTime markedAt;

	@Column(columnDefinition = "INT DEFAULT -1")
	private Integer marks;

	@Column(columnDefinition = "TEXT")
	private String feedback;

	@Column(name = "created_at", nullable = false, updatable = false)
	private LocalDateTime createdAt;

	@Column(name = "updated_at")
	private LocalDateTime updatedAt;

	@PrePersist
	protected void onCreate() {
		createdAt = LocalDateTime.now();
		updatedAt = LocalDateTime.now();
	}

	@PreUpdate
	protected void onUpdate() {
		updatedAt = LocalDateTime.now();
	}
}

File: lms-backend/src/models/Submission.js
const { DataTypes } = require("sequelize");

module.exports = (sequelize) => {
  return sequelize.define("Submission", {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    assignmentId: { type: DataTypes.INTEGER },
    assignmentTitle: { type: DataTypes.STRING },
    studentId: { type: DataTypes.INTEGER },
    studentName: { type: DataTypes.STRING },
    filePath: { type: DataTypes.STRING },
    submittedAt: { type: DataTypes.DATE },
    marks: { type: DataTypes.INTEGER, defaultValue: -1 },
    feedback: { type: DataTypes.TEXT },
    status: { type: DataTypes.STRING, defaultValue: "SUBMITTED" },
  });
};
